src/tink/core/Error.hx

```diff
@@ -42,11 +43,15 @@ class TypedError<T> {
 	public var code(default, null):ErrorCode;
 	public var data(default, null):T;
 	public var pos(default, null):Null<Pos>;
+	public var callStack(default, null):Array<StackItem>;
+	public var exceptionStack(default, null):Array<StackItem>;
 	
 	public function new(?code:ErrorCode = InternalError, message, ?pos) {
 		this.code = code;
 		this.message = message;
 		this.pos = pos;
+		this.callStack = try CallStack.callStack() catch(e:Dynamic) [];
+		this.exceptionStack = try CallStack.exceptionStack() catch(e:Dynamic) [];

```

Comment:
I don't know whether we ca detect the `-debug` flag inside haxe code or not. If yes, that should be the most convenient solution.